zubhub_frontend/zubhub/src/views/create_activity/script.js

```diff
@@ -10,787 +10,722 @@ const idPrefix = 'activitystep';
-        formData['images'] = [...(images || []), ...uploadedImages.images.map(item => ({ image: replaceImageUrlWithFileUrl(item) }))]
-
-        if (materials_used_image) {
-            formData['materials_used_image'] = {
-                file_url: materials_used_image.files[0].image_url,
-                public_id: materials_used_image.files[0].public_id
-            }
-        }
-
-        if (!materials_used_image && uploadedImages.materials_used_image[0]) {
-            materials_used_image = uploadedImages.materials_used_image[0]
-            formData['materials_used_image'] = {
-                file_url: materials_used_image.file_url,
-                public_id: materials_used_image.public_id
-            }
-        }
-
+    if (!materials_used_image && uploadedImages.materials_used_image[0]) {
+      materials_used_image = uploadedImages.materials_used_image[0];
+      formData['materials_used_image'] = {
+        file_url: materials_used_image.file_url,
+        public_id: materials_used_image.public_id,
+      };
+    }
 
-        // replace non uploaded images in formData fields with uploaded images
+    // replace non uploaded images in formData fields with uploaded images
 
-        // Assign making steps uploaded and formated images to their associated steps
-        if (makinStepsImages.length > 0) {
-            const makingStepsWithUploadedImages = formData.making_steps.map((stepData, index) => {
-                const uploaded = uploadedImages.making_steps[index]
-                const data = {
-                    ...stepData,
-                    image: [...makinStepsImages[index].files.map(item => replaceImageUrlWithFileUrl(item)), ... (uploaded || [])],
-                    step_order: index + 1
-                };
-                delete data.id;
-                delete data.images
-                return data;
-            })
+    // Assign making steps uploaded and formated images to their associated steps
+    if (makinStepsImages.length > 0) {
+      const makingStepsWithUploadedImages = formData.making_steps.map((stepData, index) => {
+        const uploaded = uploadedImages.making_steps[index];
+        const data = {
+          ...stepData,
+          image: [...makinStepsImages[index].files.map(item => replaceImageUrlWithFileUrl(item)), ...(uploaded || [])],
+          step_order: index + 1,
+        };
+        delete data.id;
+        delete data.images;
+        return data;
+      });
 
-            formData.making_steps = makingStepsWithUploadedImages
-        }
+      formData.making_steps = makingStepsWithUploadedImages;
     }
+  }
 
-    const activityId = props.match.params?.id
-    let createOrUpdateResponse;
+  const activityId = props.match.params?.id;
+  let createOrUpdateResponse;
 
-    if (props.match.params.id) {
-        // API call to the update activity
-        createOrUpdateResponse = await API.updateActivity(props.auth.token, activityId, formData)
+  if (activityId) {
```

Comment:
performs activity data updating and creation.

Since we have access to the activity `id` in the `url` when editting, we aim on that as the source of truth. So if the `id` is available, we call `updateActivity` else `createActivity`.
Then according to the different response statuses, we show the `toasts`